Sources/Apollo/GraphQLMap.swift

```diff
@@ -0,0 +1,19 @@
+public typealias GraphQLMap = [String: JSONEncodable?]
+
+fileprivate extension Dictionary where Key == String, Value == JSONEncodable? {
+  var withNilValuesRemoved: Dictionary<String, JSONEncodable> {
+    filter { $0.value != nil }
+  }
+}
+
+public protocol GraphQLMapConvertible: JSONEncodable {
+  var graphQLMap: GraphQLMap { get }
+}
+
+public extension GraphQLMapConvertible {
+  var jsonValue: JSONValue {
+    return graphQLMap.withNilValuesRemoved.jsonValue
+  }
+}
+
+public typealias GraphQLID = String
```

Comment:
This seems like something we should probably be defining somewhere else but given that I'm not sure where, here's fine for now.

Reply:
Yes haha. I 100% agree. It was defined with all of this stuff already, and I just moved the GraphQLMap to its own file, so I just moved it with that.